apps/mark-scan/pat-device-input/src/pat_input_reader.rs

```diff
@@ -0,0 +1,73 @@
+use std::io;
+use vx_logging::{log, EventId};
+
+use crate::pin::Pin;
+
+// Newer kernel versions (6.x) will offset pin addresses by 512.
+// eg. the is_connected pin is 478 + 512 = 990
+// Even though prod is expected to need the offset, we use the constant
+// values below because
+// 1. These are the documented values
+// 2. Starting low and adding the offset, instead of starting high and
+//    subtracting the offset, means we don't have to worry about
+//    pin addresses < 0
+const POSSIBLE_PIN_OFFSET: u16 = 512;
+const IS_CONNECTED_PIN_ADDRESS: u16 = 478;
+const SIGNAL_A_PIN_ADDRESS: u16 = 481;
+const SIGNAL_B_PIN_ADDRESS: u16 = 476;
+
+pub struct PatInputReader {
+    is_connected_pin: Pin,
+    signal_a_pin: Pin,
+    signal_b_pin: Pin,
+}
+
+impl PatInputReader {
+    pub fn new() -> Result<Self, io::Error> {
```

Comment:
I was skeptical of returning `Result` from a constructor but it seems to be an accepted pattern per https://users.rust-lang.org/t/idiomatic-error-handling-in-new-and-builder-pattern/86361 and https://users.rust-lang.org/t/should-i-return-a-result-from-new/17372